### PR TITLE
test: add sast test with semgrep

### PR DESCRIPTION
**Pull request by indecisivedragon**
- [x] Followed the steps in the contributor's guide: https://crashoverride.com/docs/other/contributing#filing-the-pull-request
- [x] PR title uses [semantic commit messages](https://nitayneeman.com/posts/understanding-semantic-commit-messages-using-git-and-angular/#fix)
- [x] Filled out the template to a useful degree

## Issue

closes https://github.com/crashappsec/chalk/issues/95

## Description

add test for semgrep

### CODE REVIEW

File: tests/data/python/sample_1/helloworld.py [hunk 1]
```diff
@@ -2,6 +2,10 @@
 # dummy comment second line
 def main():
     print("Hello World")
+    # this is added for semgrep test
+    test_var = "aaa"
+    if test_var is "bbb":
```
Comment (miki725) [inline tests/data/python/sample_1/helloworld.py:7]:
```suggestion
    if test_var == "bbb":
```

`is` compares singletons which strings are not guaranteed to be singleton

python internally caches small strings so in most cases its true but its def not true for long strings:

```python
>>> "a" is "a"
<stdin>:1: SyntaxWarning: "is" with a literal. Did you mean "=="?
True

>>> "a" * 2**20 is "a" * 2**20
False
```

File: tests/test_plugins.py [hunk 2]
```diff
@@ -712,3 +713,120 @@ def test_syft_binary(copy_files: list[Path], chalk_copy: Chalk):
...
+    # check that sbom has been embedded into the artifact
+    chalk_mark = ChalkMark.from_binary(tmp_data_dir / "helloworld.py")
+    assert chalk_mark.contains(sast_data)
+
+    # logger.info("after insert")
+
+    # text = (tmp_data_dir / "helloworld.py").read_text()
+    # logger.info(text)
+
+    # time.sleep(1000)
```
Comment (miki725) [inline tests/test_plugins.py:832]: forgot to cleanup?

Comment (indecisivedragon) [reply tests/test_plugins.py]: yup, whoops

File: tests/data/python/sample_1/helloworld.py (hunk 1 above)
Comment (indecisivedragon) [reply tests/data/python/sample_1/helloworld.py:7]: we need this to be `is` for the semgrep rule to trigger correctly (`rule,yaml`)

**Review by viega — APPROVED** [review]

**Pull request merged**